### PR TITLE
Fix push format

### PR DESCRIPTION
**Pull request by andreaskuster**

### CODE REVIEW

**Comment by andreaskuster** [comment]
It seems as if it is not possible to trigger an action from a commit of a previous action (to prevent infinity-loops, ..) see: https://peterevans.dev/posts/github-actions-how-to-automate-code-formatting-in-pull-requests/

Therfore, we will let the pipeline fail in case of not passing clang-format (another option would be to open a new pull request, but i think that this solution here is better).

**Review by KarlSchu — COMMENTED** [review]
What about leave out `.pio` in formatting?
All:  178 files in `.pio` but 44 files in `./lib` only

```bash
find . -name "*.cpp" -o -name "*.c" -o -name "*.h" | grep '\./include' | xargs -I {} clang-format -i {}
find . -name "*.cpp" -o -name "*.c" -o -name "*.h" | grep '\./interfaces' | xargs -I {} clang-format -i {}
find . -name "*.cpp" -o -name "*.c" -o -name "*.h" | grep '\./lib' | xargs -I {} clang-format -i {}
find . -name "*.cpp" -o -name "*.c" -o -name "*.h" | grep '\./src' | xargs -I {} clang-format -i {}
find . -name "*.cpp" -o -name "*.c" -o -name "*.h" | grep '\./text' | xargs -I {} clang-format -i {}
```

**Comment by andreaskuster** [comment]
> 
> 
> What about leave out `.pio` in formatting?
> All: 178 files in `.pio` but 44 files in `./lib` only
> 
> ```shell
> find . -name "*.cpp" -o -name "*.c" -o -name "*.h" | grep '\./include' | xargs -I {} clang-format -i {}
> find . -name "*.cpp" -o -name "*.c" -o -name "*.h" | grep '\./interfaces' | xargs -I {} clang-format -i {}
> find . -name "*.cpp" -o -name "*.c" -o -name "*.h" | grep '\./lib' | xargs -I {} clang-format -i {}
> find . -name "*.cpp" -o -name "*.c" -o -name "*.h" | grep '\./src' | xargs -I {} clang-format -i {}
> find . -name "*.cpp" -o -name "*.c" -o -name "*.h" | grep '\./text' | xargs -I {} clang-format -i {}
> ```

I think it is cleaner to just ignore all hidden directories instead.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,10 +1,23 @@
...
+  format:
+    name: Clang-Format
+    runs-on: ubuntu-latest
+    steps:
+      - name: install
+        run: sudo apt-get install -y clang-format
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: format
+        run: find . -not -path "./.*" -name "*.cpp" -o -name "*.c" -o -name "*.h" | xargs -I {} clang-format -i {}
```
Comment (KarlSchu) [inline .github/workflows/ci.yml:16]: @andreaskuster Proposel
find . -type d \( -path "./.*" -o -path "./datasheets" -o -path "./medis" \) -prune -false -o -name "*.h" -o -name ".hpp" -o -name "*.cpp" | xargs -I {} clang-format -i {}
It is the same as in [https://github.com/SolarEnergyRacers/solar_car_control_system/blob/convert_flat_c_to_cpp_classes/extras/format.sh](https://github.com/SolarEnergyRacers/solar_car_control_system/blob/convert_flat_c_to_cpp_classes/extras/format.sh)

Comment (andreaskuster) [reply .github/workflows/ci.yml:16]: Sure, I am fine with that. Please commit it and merge the PR.

**Review by KarlSchu — APPROVED** [review]

**Pull request merged**